### PR TITLE
Fix missing schedule name

### LINKED ISSUE #797: Zimfarm at youzim.it doesn't show schedule names
[//]: # (Pick appropriate classifiers)
- Location: __U.I.__
- Schedule Name/ID: __n/a__
- Task ID: __n/a__

## Problem

image

## Reproducing steps

Go to https://farm.youzim.it/pipeline/filter-done and observe no schedule names.

### PR DESCRIPTION
**Pull request by benoit74**
## Rationale

- Fix #797 
- some small fixes

## Changes
- a new log line is writen when a CMS error occurs, so that we can search for it in Grafana ("CMS errors" indicator is now broken since we do not have the exception anymore since https://github.com/openzim/zimfarm/pull/827
- fix a test issues where tests where not re-entrant
- add a `frontend-dev-ui` development container with hot reload of development changes, very useful for UI development
- store in DB `original_schedule_name` in `requested_tasks` and `tasks`
- expose the `original_schedule_name` to the API along the `schedule_name` which will remain empty in youzimit
- adapt UI to a missing `schedule_name` and use the original one.
- add a Jupyter notebook to test code e.g. SQLAlchemy query + move existing one in `dev` folder which is more appropriate

### CODE REVIEW

File: dev/docker-compose.yml [hunk 1]
```diff
@@ -41,6 +41,20 @@ services:
       ZIMFARM_WEBAPI: http://localhost:8000/v1
     depends_on:
       - backend
+  frontend-ui-dev:
+    build:
+      dockerfile: ../../dev/frontend-ui-dev/Dockerfile
+      context: ../dispatcher/frontend-ui
+    # build: ../dispatcher/frontend-ui
```
Comment (rgaudin) [inline dev/docker-compose.yml:48]: we can remove that

File: dev/frontend-ui-dev/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+RUN apk --no-cache add yarn python3
+WORKDIR /app
+COPY entrypoint.sh /usr/local/bin/entrypoint.sh
+COPY package.json yarn.lock /app/
+RUN yarn install && yarn cache clean
+COPY *.js /app/
+COPY public /app/public
+COPY src /app/src
+#CMD ["sleep", "infinity"]
```
Comment (rgaudin) [inline dev/frontend-ui-dev/Dockerfile:11]: that too

File: dispatcher/frontend-ui/entrypoint.sh [hunk 3]
```diff
@@ -1,11 +1,14 @@
 #!/bin/sh
 
-JS_PATH=/usr/share/nginx/html/environ.json
-echo "dump ZIMFARM_* environ variables to $JS_PATH"
+if [ -z "${ENVIRON_PATH}" ]
```
Comment (rgaudin) [inline dispatcher/frontend-ui/entrypoint.sh:3]: I think we'd better set it in both Dockerfile and minimize this

**Review by rgaudin — APPROVED** [review]
Thank you ; I think we can skip the condition in entrypoint.

**Comment by benoit74** [comment]
Thank you, I've fixed and resolved obvious conversation
I let you check the change regarding ENVIRON_PATH and merge once ok.

**Review by rgaudin — APPROVED** [review]
LGTM 👍

**Pull request merged**